source/fundamentals/connection.txt

```diff
@@ -0,0 +1,636 @@
+
+   .. code-block:: xml
+
+      <resource-ref>
+        <res-ref-name>
+          mongodb/MyMongoClient
+        </res-ref-name>
+        <res-type>
+          com.mongodb.MongoClient
+        </res-type>
+        <res-auth>
+          Container
+        </res-auth>
+      </resource-ref>
+
+This will make a ``MongoClient`` instance accessible via the JNDI name
+``mongodb/MyMongoClient`` in the ``java:comp/env`` context.
+
+.. _tls:
+
+TLS/SSL
+-------
+
+You can connect to MongoDB instances with the
+:wiki:`TLS/SSL </Transport_Layer_Security>` security protocol using
+the underlying TLS/SSL support in the JDK. To configure your connection
+to use TLS/SSL, enable the TLS/SSL settings in either the
+:java-sync-api:`ConnectionString
+<com/mongodb/ConnectionString.html>` or :java-sync-api:`MongoClientSettings
+<com/mongodb/MongoClientSettings.html>` objects.
+
+.. _tls-connectionstring:
+
+Enable TLS/SSL via ConnectionString
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable TLS/SSL with a :java-sync-api:`ConnectionString
+<com/mongodb/ConnectionString.html>`, assign the connection string
+parameter ``ssl`` a value of ``true`` in the connection string passed to
+``MongoClients.create()``:
+
+.. code-block:: java
+
+   MongoClient mongoClient = MongoClients.create("mongodb://localhost/?ssl=true");
+
+.. _tls-mongoclientsettings:
+
+Enable TLS/SSL via MongoClientSettings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To enable TLS/SSL with :java-sync-api:`MongoClientSettings
+<com/mongodb/MongoClientSettings.html>`, set the ``enabled``
+property of the builder to ``true`` using the
+:java-sync-api:`applyToSslSettings()
+<com/mongodb/MongoClientSettings.Builder.html#applyToSslSettings(com.mongodb.Block)>`
+builder block:
+
+.. code-block:: java
+   :emphasize-lines: 2-3
+
+   MongoClientSettings settings = MongoClientSettings.builder()
+        .applyToSslSettings(builder -> 
+            builder.enabled(true))
+        .build();
+   MongoClient client = MongoClients.create(settings);
+
+.. _tls-disable-hostname-verification:
+
+Disable Hostname Verification
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the driver ensures that the hostname included in the server's
+SSL certificate(s) matches the hostname(s) provided when constructing a
+``MongoClient``. If your application needs to disable hostname
+verification, you can explicitly disable this by setting the
+``invalidHostNameAllowed`` property of the builder to true in the
+``applytoSslSettings()`` builder modifier:
+
+.. code-block:: java
+   :emphasize-lines: 4
+
+   MongoClientSettings settings = MongoClientSettings.builder()
+        .applyToSslSettings(builder -> {
+                    builder.enabled(true);
+                    builder.invalidHostNameAllowed(true);
+                })
+        .build();
+
+.. _tls-configure-jvm-truststore:
+
+Configure the JVM TrustStore
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Most applications need to configure two JVM system properties to ensure
```

Comment:
With the restructured two-section layout we have now, I'm going to leave the numbers there -- store/StorePassword is a pattern that's remained unchanged for a long time in the JVM, and I doubt it'll change any time soon. In this case I prefer to be straightforward with numbers since omitting them always makes me (as a reader) wonder how much more I'll have to do when I scroll down further!

Reply:
The issue is that you mention "Most applications need to configure two properties" and later "the typical application also needs to set two properties", not that I think that keystore and truststore are going to change in the JVM. I know you're addressing"typical application", doesn't "most" imply some overlap with "typical" and therefore four are usually required?

Reply:
I've pushed my changes, and I feel that separating the key/trust store sections (under one larger Configure Certificates section) has made things a little more goal-oriented and hopefully addressed the issue. If you feel strongly about it I can remove the numbers entirely, though.